### PR TITLE
Fix rubocop warnings

### PR DESCRIPTION
**Pull request by paulomarg**
This PR fixes (or ignores) rubocop warnings on the codebase, and removes the associated ignores from the TODO file.

### CODE REVIEW

File: lib/shopify_api/resources/line_item.rb [hunk 1]
```diff
@@ -2,11 +2,11 @@ module ShopifyAPI
...
     class Property < Base
       def initialize(*args)
-        attributes = args[0] || {}
-        persisted = args[1] || false
+        self.attributes = args[0] || {}
+        self.persisted = args[1] || false
         super
       rescue NameError
-        attributes = attributes.to_hash
+        self.attributes = attributes.to_hash
```
Comment (paulomarg) [inline lib/shopify_api/resources/line_item.rb:9]: I wasn't sure whether to do this or ignore this particular issue, but it doesn't seem like these variables were used at all in the previous code.

Comment (andyw8) [reply lib/shopify_api/resources/line_item.rb:9]: Yeah, it seems flawed.

File: test/price_rule_test.rb [hunk 2]
```diff
@@ -43,7 +43,7 @@ def test_create_price_rule
     )
 
     assert_equal '{"price_rule":{"target_type":"line_item","allocation_method":"across","value_type":"fixed_amount","value":-10.0,"customer_selection":"all","starts_at":"2017-01-19T00:00:00Z"}}', WebMock.last_request.body
-    assert_equal -10, price_rule.value
+    assert_equal(-10, price_rule.value)
```
Comment (andyw8) [inline test/price_rule_test.rb:46]: I don't understand why it's correcting this but not the other `assert_equal` calls?

File: test/session_test.rb [hunk 3]
```diff
@@ -519,7 +519,7 @@ def make_sorted_params(params)
 
   def generate_signature(params)
     params = make_sorted_params(params) if params.is_a?(Hash)
-    OpenSSL::HMAC.hexdigest(OpenSSL::Digest::SHA256.new, ShopifyAPI::Session.secret, params)
+    OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('SHA256'), ShopifyAPI::Session.secret, params)
```
Comment (andyw8) [inline test/session_test.rb:522]: I would be cautious of how this might impact different Ruby versions.

Comment (paulomarg) [reply test/session_test.rb:522]: We run the tests for every ruby version between 2.4 and 2.7, and the CI build worked. Is there anything else in particular we should look for?

File: test/price_rule_test.rb (hunk 2 above)
Comment (paulomarg) [reply test/price_rule_test.rb:46]: This one in particular complained about the `-10` as it's not clear whether it's a parameter or not. We have ~600 infractions on the check for calling methods without parameters, but that's a lower severity warning. We'll get there 😄

File: test/session_test.rb (hunk 3 above)
Comment (andyw8) [reply test/session_test.rb:522]: Ok, then it should be fine. I'm just cautious of OpenSSL related things being fragile, based on past bad experience!

**Review by andyw8 — APPROVED** [review]

**Pull request merged**